packages/celotool/src/e2e-tests/cip35_tests.ts

```diff
@@ -0,0 +1,454 @@
+    this.kitWithLocalWallet.connection.addAccount(validatorPrivateKey)
+    this.kitWithLocalWalletLight.connection.addAccount(validatorPrivateKey)
+  }
+
+  async generateUnprotectedTransaction(ethCompatible: boolean): Promise<string> {
+    const encode = (ejsUtil as any).rlp.encode // the typescript typings are incomplete
+    const numToHex = (x: number | BigNumber) => ejsUtil.bufferToHex(ejsUtil.toBuffer(x))
+    const nonce = await this.kit.connection.nonce(validatorAddress)
+    const celoOnlyFields = ethCompatible ? [] : ['0x', '0x', '0x']
+    const arr = [
+      nonce > 0 ? numToHex(nonce) : '0x',
+      numToHex(parseInt(this.gasPrice, 10)),
+      numToHex(1000000), // plenty of gas
+      ...celoOnlyFields,
+      toAddress, // to
+      '0x05', // value: 5 wei
+      '0x', // no data
+    ]
+    const signingHash = ejsUtil.rlphash(arr)
+    const pk = ejsUtil.addHexPrefix(validatorPrivateKey)
+    const sig = ejsUtil.ecsign(signingHash, ejsUtil.toBuffer(pk))
+    arr.push(ejsUtil.bufferToHex(sig.v), ejsUtil.bufferToHex(sig.r), ejsUtil.bufferToHex(sig.s))
+    return ejsUtil.bufferToHex(encode(arr))
+  }
+
+  runReplayProtectionTests() {
+    for (const ethCompatible of [false, true]) {
+      this.runReplayProtectionTest(ethCompatible)
+    }
+  }
+
+  runReplayProtectionTest(ethCompatible: boolean) {
+    describe(`Transaction without replay protection, ethCompatible: ${ethCompatible}`, () => {
+      let minedTx: any = null // Use any because we haven't added `ethCompatible` to these types
+      let error: string | null = null
+
+      before(async () => {
+        const tx = await this.generateUnprotectedTransaction(ethCompatible)
+        try {
+          const receipt = await (await this.kit.connection.sendSignedTransaction(tx)).waitReceipt()
+          minedTx = await this.kit.web3.eth.getTransaction(receipt.transactionHash)
+          error = null
+        } catch (err) {
+          error = err.message
+        }
+      })
+
+      if (ethCompatible && !this.cipIsActivated) {
+        it('fails due to being ethereum-compatible', () => {
+          assert.isNull(minedTx, 'Transaction succeeded when it should have failed')
+          assert.match(
+            error!,
+            new RegExp(notYetActivatedError),
+            `Got "${error}", expected "${notYetActivatedError}"`
+          )
+        })
+      } else if (this.cipIsActivated) {
+        // Replay protection is mandatory, so the transaction should fail
+        it('fails due to replay protection being mandatory', () => {
+          assert.isNull(minedTx, 'Transaction succeeded when it should have failed')
+          assert.match(
+            error!,
+            new RegExp(noReplayProtectionError),
+            `Got "${error}", expected "${noReplayProtectionError}"`
+          )
+        })
+      } else {
+        // Should succeed, since replay protection is optional before Donut
+        it('succeeds', () => {
```

Comment:
Are we sure about this? eip155 was already implemented. And I recall having problems it we didn't add that to the signature

Reply:
Yes, it works. See https://github.com/celo-org/celo-blockchain/blob/68e2ab747b40ce215a916cde7fdc77613366dcd0/core/types/transaction_signing.go#L128-L130

I'm don't know what trouble you ran into, probably the raw transaction wasn't being prepared correctly (which is a problem I ran into several times in working on this test)